### PR TITLE
[GPU]  optimize permute_ref 

### PR DESCRIPTION
**Pull request by yeonbok**
### Details:
 - Fix access order of permute_ref to process f axis first if input is fsv and output is bfyx and the f is moved to x position

### Tickets:
 - 86967

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/permute/permute_kernel_ref.cpp [hunk 1]
```diff
@@ -33,14 +34,44 @@ CommonDispatchData PermuteKernelRef::SetDefault(const permute_params& params) co
...
     auto in_layout = params.inputs[0].GetLayout();
     auto out_layout = params.outputs[0].GetLayout();
-    std::vector<std::vector<Tensor::DataChannelName>> dims_by_gws = {{Tensor::DataChannelName::X},
-                                                                     {Tensor::DataChannelName::Y, Tensor::DataChannelName::Z, Tensor::DataChannelName::W},
-                                                                     {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
 
-    const auto& in =  params.inputs[0];
+    std::function<bool(const std::vector<uint16_t>&)> f_to_x = [](const std::vector<uint16_t>& order) {
+        if ((int32_t) order[1] == order.size() - 1) return true;
+        return true;
```
Comment (si-eun-kim) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/permute/permute_kernel_ref.cpp:40]: I see return true for both return statements. Maybe your intension was 'return false;' for one of the two return statements?

**Review by ahnyoung-paul — APPROVED** [review]
LGTM

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/permute/permute_kernel_ref.cpp [hunk 2]
```diff
@@ -29,18 +29,53 @@ ParamsKey PermuteKernelRef::GetSupportedKey() const {
...
+            case Tensor::DataLayout::b_fs_yx_fsv4:
+            case Tensor::DataLayout::b_fs_yx_fsv16:
+            case Tensor::DataLayout::b_fs_yx_fsv32:
+            case Tensor::DataLayout::b_fs_zyx_fsv16:
+            case Tensor::DataLayout::b_fs_zyx_fsv32:
+                return true;
+            default:
+                return false;
+        }
+        return false;
```
Comment (e-ddykim) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/permute/permute_kernel_ref.cpp:51]:
```suggestion
```

Comment (e-ddykim) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/permute/permute_kernel_ref.cpp:51]: L51 can be removed. But it's not a problem.

**Review by e-ddykim — APPROVED** [review]
Looks good to me

**Pull request merged**